### DISCUSSION TITLE
authRecord is null on 'onFileDownloadRequest'

### DISCUSSION THREAD

**Discussion by pagoru** [pocketbase/pocketbase#4629] [Q&A] [answered]
I'm trying to log when someone downloads a file, but I'm having trouble to identify the current authenticated user. 
I checked that the collection is protected with `@request.auth.id != ""` and It works fine (and only for logged users), but when I use `onFileDownloadRequest`, the authRecord is always null.
I'm doing something wrong or there's a bug? Thanks!

```ts
onFileDownloadRequest((e) => {
  const data = $apis.requestInfo(e.httpContext);

  //always null
  console.log(data.authRecord);
});
```

**Comment by ganigeorgiev** [comment, marked as answer]
How you are sending the file request?

`onFileDownloadRequest` hook is triggered when you request to download/access a file. If you are accessing the file via `img` tag or directly from the browser url bar then the request will not have `Authorization` header and no information about the auth state. That is why short-lived file token as query param was introduced for the cases when the file needs to be "protected" - https://pocketbase.io/docs/files-handling/#protected-files.

**Reply by pagoru** [reply]
Ohhh, now I understand. I was using `window.location.assign` but it's the same, of course!
I can append the header + check if the header is present on the hook, and It will work!

Thanks! 👯

**Reply by pagoru** [reply]
I already tested that, works perfect! Thanks mate
